src/main/java/com/deviceinsight/kafka/health/KafkaConsumingHealthIndicator.java

```diff
@@ -74,21 +81,38 @@ public KafkaConsumingHealthIndicator(KafkaHealthProperties kafkaHealthProperties
 		this.consumer = new KafkaConsumer<>(kafkaConsumerPropertiesCopy, deserializer, deserializer);
 		this.producer = new KafkaProducer<>(kafkaProducerProperties, serializer, serializer);
 
-		this.executor = new ThreadPoolExecutor(0, 1, 0L, MILLISECONDS, new SynchronousQueue<>(),
-				new ThreadPoolExecutor.AbortPolicy());
+		this.executor = Executors.newFixedThreadPool(2);
+		this.running = new AtomicBoolean(true);
+		this.cacheService = new CaffeineCacheServiceImpl(calculateCacheExpiration(sendReceiveTimeoutMs));
+
+		this.kafkaCommunicationResult = KafkaCommunicationResult.failure(topic, new RejectedExecutionException("Kafka Health Check is starting."));
 	}
 
 	@PostConstruct
 	void subscribeAndSendMessage() throws InterruptedException {
 		subscribeToTopic();
-		KafkaCommunicationResult kafkaCommunicationResult = sendAndReceiveMessage();
+
+		sendMessage();
+
 		if (kafkaCommunicationResult.isFailure()) {
 			throw new RuntimeException("Kafka health check failed", kafkaCommunicationResult.getException());
 		}
+
+		executor.submit(() -> {
+			while (running.get()) {
+				if (messageNotReceived()) {
+					this.kafkaCommunicationResult = KafkaCommunicationResult.failure(topic,
+							new RejectedExecutionException("Ignore health check, already running..."));
```

Comment:
I don't understand why this error message is used here.

Reply:
That's the method where we continuous polling. If nobody asks for the Health Check, no message will be written on the topic and therefore he won't be able to consume here.

Reply:
But then the message is still misleading. I would prefer something like "No message received", because that's what happens.